preloader.go

```diff
@@ -0,0 +1,74 @@
+/*
+ *
+ * Copyright 2014 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package grpc
+
+import (
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	// "google.golang.org/grpc/encoding"
+)
+
+// PreparedMsg is responsible for creating a Marshalled and Compressed object
+type PreparedMsg struct {
+	// Struct for preparing msg before sending them
+	encodedData []byte
+	compredData []byte
+	hdr         []byte
+	payload     []byte
+}
+
+// checks if the rpcInfo has all the correct information
+func checkPreparedMsgContext(rpc *rpcInfo) error {
```

Comment:
It is only used in Encode. Why this check is factored out into a function?

Reply:
On second thought, I believe it should be exported  as 
```go
CheckPreparedMsgContext(*rpcInfo) error
```
because an implementation might want to confirm if they have prepared the context properly.

Reply:
In that case I will refactor it, and remove the separate function.